### DISCUSSION TITLE
Codespace secrets not being passed into container

### DISCUSSION THREAD

**Discussion by ethanteague** [community/community#43626] [Codespaces]
### Select Topic Area

Bug

### Body

I need to authenticate with an external service (Acquia cloud), and have my key stored in secrets as `ID_RSA`. When I run a codespace, and do `echo env` I see all the expected variables relating to my user account, etc., but secret variables do not appear (e.g., `ID_RSA` does not appear as a named variable). I was able to access this secret up until a couple of weeks ago (granted, I didn't look for it via the echo env method while it was working). So, I'm not sure if this is a bug, or if I am unaware of the correct method for accessing a variable to be made available to my container in devcontainer.json.

**Comment by aditya30394** [comment]
Thanks for bringing it to our notice. We haven't changed anything recently so this should work. Have you followed the directions mentioned in this doc ? - https://docs.github.com/en/codespaces/managing-your-codespaces/managing-encrypted-secrets-for-your-codespaces

**Comment by rayjlinden** [comment]
This is not working for me either.  (Seems to work fine in actions so I'm rather confused.). I'm sure I'm doing something stupid...  Were you able to get it working?  What was your fix?

**Comment by keenan-v1** [comment]
For anyone having this issue with secrets specifically named `*_CONTAINER_REGISTRY_*`, I've found that these secrets are handled in a special way and are NOT available in the environment.

**Comment by rayjlinden** [comment]
And is there any docs that describe that codespaces will delete the values of *_CONTAINER_REGISTRY_* environment variables?  While we now know that is what is going on I still have yet to come across any documentation of this behavior.

**Comment by r-uu** [comment]
still not working for me either in november 2024. any suggestions for workarounds?
